lxd-agent/network.go

```diff
@@ -69,3 +81,139 @@ func serverTLSConfig() (*tls.Config, error) {
 
 	return tlsConfig, nil
 }
+
+// reconfigureNetworkInterfaces checks for the existence of files under ./nics in the config share.
+// Each file is named <device>.json and contains the MTU and MAC address.
+func reconfigureNetworkInterfaces() {
+	// Abort if configuration folder does not exist (nothing to do), otherwise log and return.
+	nicDir, err := ioutil.ReadDir("nics")
+	if os.IsNotExist(err) {
+		return
+	} else if err != nil {
+		logger.Error("Could not read network interface configuration directory", log15.Ctx{
+			"error": err,
+		})
+		return
+	}
+
+	// nicData is a map of MAC address to new interface name and MTU.
+	nicData := make(map[string]struct {
+		Name string
+		MTU  uint32
+	})
+	for _, f := range nicDir {
+		nicBytes, err := ioutil.ReadFile(filepath.Join("nics", f.Name()))
+		if err != nil {
+			logger.Error("Could not read network interface configuration file", log15.Ctx{
+				"error": err,
+			})
+		}
+
+		var conf deviceConfig.NICConfig
+		err = json.Unmarshal(nicBytes, &conf)
+		if err != nil {
+			logger.Error("Could not parse network interface configuration file", log15.Ctx{
+				"error": err,
+			})
+			return
+		}
+
+		// The new interface name will be the device name (which is the file name sans ".json").
+		newInterfaceName := strings.TrimSuffix(f.Name(), filepath.Ext(f.Name()))
+		nicData[conf.MACAddress] = struct {
+			Name string
+			MTU  uint32
+		}{
+			Name: newInterfaceName,
+			MTU:  conf.MTU,
+		}
+	}
+
+	configureNIC := func(currentInterfaceName, currentMACAddress, currentMTU string) error {
+		reverter := revert.New()
+		defer reverter.Fail()
+
+		nic, ok := nicData[currentMACAddress]
+		if !ok {
+			return nil
+		}
+
+		_, err = shared.RunCommand("ip", "link", "set", "down", currentInterfaceName)
```

Comment:
I wonder, could we use the `lxd/ip` for all this stuff instead of directly using shared.RunCommand here?